### ISSUE TITLE
bug(migration): Invalid import after migration

### ISSUE THREAD

**Issue by mcalmus** [angular/components#25672]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Migration from Material 14 to Material 15 left a compile error

### Reproduction

Steps to reproduce:
1. Create an Angular/Material 14 project
2. Have MatOptionSelectionChange import in some component
3. Migrate to Angular/Material 15

### Expected Behavior

Code compiles

### Actual Behavior

Compile error:

```
'"@angular/material/legacy-core"; has no exported member named 'MatLegacyOptionSelectionChange'. Did you mean 'MatOptionSelectionChange'?

import {MatLegacyOptionSelectionChange as MatOptionSelectionChange } from '@angular/material/legacy-core';
```

### Environment

- Angular: 15.0.0-next.2
- CDK/Material: 15.0.0-next.1
- Browser(s): n/a
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by json-derulo** [comment]
fixed by #25673

**Comment by crisbeto** [comment]
Closing as fixed.
